torch/testing/_creation.py

```diff
@@ -203,6 +203,11 @@ def clamp(a: float, l: float, h: float) -> float:
                 default_high=10,
             ),
         )
+        if low + 1 == high:
+            raise ValueError(
+                f"Computed low ({low}) and high ({high}) values are almost equal. "
+                "This will generate a constant tensor which may lead to potential silent bugs"
+            )
```

Comment:
I don't think we should have this error. `make_tensor` is consistent with other random tensor creation ops, i.e. treating `low` as inclusive and `high` as exclusive:

```python
low = 3
a = torch.testing.make_tensor((3, 3), low=low, high=low + 1, dtype=torch.int, device="cpu")
b = torch.randint(size=(3, 3), low=low, high=low + 1)

assert (a == b).all()
```

I agree, this might hide a bug, but I don't think we make the situation any less confusing by diverging from other well-known ops.

I'm more open to a discussion about a warning here. Since this is a testing function, that shouldn't create too much unwanted noise.

Reply:
It also looks like the case in `sample_inputs_multi_margin_loss` is intentionally creating a constant tensor and just using `make_tensor` for symmetry with the other samples.

Reply:
IMO, `make_target([], low=0, high=1)` should be just `torch.tensor(0, device=device)` to be clear what the sample is.

Reply:
Reverted this change